### PR TITLE
Add get_max_threads_per_block method to Dispatcher

### PR DESCRIPTION
**Pull request by testhound**
This pull request partially address issue: https://github.com/numba/numba/issues/6448 by adding support for retrieving the maximum threads per block utilized from the Dispatcher.

### CODE REVIEW

**Comment by testhound** [comment]
gpuci run tests

**Review by gmarkall — CHANGES_REQUESTED** [review]
Thanks for the PR. Some small comments on the diff to address, and a merge from master needed.

**Comment by testhound** [comment]
@gmarkall I updated the PR with all your recommendations.

**Comment by gmarkall** [comment]
@testhound Do you know what is going on with the tests?

**Comment by testhound** [comment]
@gmarkall I had an error in kernet.rst that is now fixed.

**Comment by testhound** [comment]
gpuci run tests

File: numba/cuda/dispatcher.py [hunk 1]
```diff
@@ -777,6 +784,27 @@ def get_shared_mem_per_block(self, signature=None):
             return {sig: overload.shared_mem_per_block
                     for sig, overload in self.overloads.items()}
 
+    def get_max_threads_per_block(self, signature=None):
+        '''
+        Returns the maximum allowable number of threads per block
+        for this kernel. Exceeding this threshold will result in
+        the kernel failing to launch
```
Comment (gmarkall) [inline numba/cuda/dispatcher.py:791]:
```suggestion
        the kernel failing to launch.
```

**Review by gmarkall — CHANGES_REQUESTED** [review]
Many thanks for the updates. One small suggestion on the diff, then I think this should be good to merge.

**Comment by testhound** [comment]
gpuci run tests

**Review by gmarkall — APPROVED** [review]
Thanks for the update!

**Pull request merged**